src/Microsoft.TemplateEngine.Utils/DictionaryExtensions.cs

```diff
@@ -73,5 +73,44 @@ public enum ConflictingKeysResolution
                 }
             }
         }
+
+        /// <summary>
+        /// Adds a content to given dictionary if specified value satisfies condition.
+        /// </summary>
+        /// <typeparam name="TKey"></typeparam>
+        /// <typeparam name="TValue"></typeparam>
+        /// <param name="dict">Dictionary that will contained specified key/value.</param>
+        /// <param name="key">Dictionary key.</param>
+        /// <param name="value">Dictionary value.</param>
+        /// <param name="condition">Insertion condition.</param>
+        /// <exception cref="Exception">Thrown if key is already present in current dictionary and <see cref="ConflictingKeysResolution.Throw"/> strategy was requested.</exception>
+        public static bool TryAdd<TKey, TValue>(this IDictionary<TKey, TValue> dict, TKey key, TValue value, Predicate<TValue> condition)
+        {
+            if (condition(value))
+            {
+                dict[key] = value!;
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Adds a content to given dictionary if specified value satisfies condition; otherwise removes the key from the dictionary.
+        /// </summary>
+        /// <typeparam name="TKey"></typeparam>
+        /// <typeparam name="TValue"></typeparam>
+        /// <param name="dict">Dictionary that will contained specified key/value.</param>
+        /// <param name="key">Dictionary key.</param>
+        /// <param name="value">Dictionary value.</param>
+        /// <param name="condition">Insertion condition.</param>
+        /// <exception cref="Exception">Thrown if key is already present in current dictionary and <see cref="ConflictingKeysResolution.Throw"/> strategy was requested.</exception>
+        public static void UpdateOrRemoveValue<TKey, TValue>(this IDictionary<TKey, TValue> dict, TKey key, TValue? value, Predicate<TValue> condition)
+        {
+            if (value is null || !dict.TryAdd(key, value, condition))
+            {
+                dict.Remove(key);
+            }
+        }
```

Comment:
imo, this method is very specific and should stay internal to the use.

Reply:
If we switch it to `internal` we cannot access it on the `Edge` project where it is being primarily used. We can either leave it public or make it a private method on the only class that is currently using this method.

Reply:
that's what I referred to: a  private method on the only class that is currently using this method